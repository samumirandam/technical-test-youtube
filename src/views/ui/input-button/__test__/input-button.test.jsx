/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { render, screen } from '@utils/test-utils';

import InputButton from '../index';

const defaultProps = { labelButton: 'Press' };

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<InputButton {...setupStore} />, {});
};

describe('Test for InputButton ui component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByText('Press')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
