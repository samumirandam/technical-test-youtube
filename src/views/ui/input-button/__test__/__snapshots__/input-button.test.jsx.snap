// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for InputButton ui component Should renders the same component 1`] = `
<div>
  <div
    class="InputButton"
  >
    <input
      class="InputButton__input"
      placeholder=""
      type="text"
    />
    <button
      class="Button InputButton__button primary secondary"
      type="button"
    >
      Press
    </button>
  </div>
</div>
`;
